### ISSUE TITLE
SES 'VerificationStatus' stuck 'Pending'

### ISSUE THREAD

**Issue by jeff1evesque** [ImmobilienScout24/aws-cf-verified-ssl-certificate#14]
I used your codebase (cloudformation, lambda) verbatim, with the exception that instead of using `HostedZoneName`, I used the `HostedZoneId`. Additionally, I downloaded the `requests` python package in AWS Linux AMI using `pip` from python3 (not from python `pip` 2.7). The code seems to work partially/halfway. Cloudformation stack/build logs only tell me to look into the CloudWatch Logs, with the following CloudWatch logs:

```
02:32:34
RESULT: {'okcool.com': {u'VerificationToken': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx=', u'VerificationStatus': 'Pending'}}
02:32:34
==========================================================
02:32:34
Status: Pending
02:32:39
==========================================================
02:32:39
RESULT: {'okcool.com': {u'VerificationToken': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx=', u'VerificationStatus': 'Pending'}}
02:32:39
==========================================================
02:32:39
Status: Pending
02:32:44
==========================================================
02:32:44
RESULT: {'okcool.com': {u'VerificationToken': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx=', u'VerificationStatus': 'Pending'}}
02:32:44
==========================================================
02:32:44
Status: Pending
02:32:49
==========================================================
02:32:49
RESULT: {'okcool.com': {u'VerificationToken': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx=', u'VerificationStatus': 'Pending'}}
02:32:49
==========================================================
02:32:49
Status: Pending
02:32:54
==========================================================
02:32:54
RESULT: {'okcool.com': {u'VerificationToken': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx=', u'VerificationStatus': 'Pending'}}
02:32:54
==========================================================
02:32:54
Status: Pending
02:32:59
========================================================== 
```

I suppose I lied, the above logs were generated by adding additional `print` statements to your `wait_for_ses_domain_verification` method:

```python
    def wait_for_ses_domain_verification(self):
        start = datetime.now()
        status = 'Failed'
        while (datetime.now() - start).total_seconds() < 240:
            result = self.ses.get_identity_verification_attributes(Identities=[self.domain])['VerificationAttributes']
            print('==========================================================')
            print('RESULT: {a}'.format(a=str(result)))
            print('==========================================================')
            if self.domain in result:
                status = result[self.domain]['VerificationStatus']
                print 'Status: ' + status
                if status == 'Success':
                    break
                time.sleep(5)
        if status != 'Success':
            raise Exception('Verification took to long. Aborting...')
```

To sum, it seems the above method cannot find any trace that the corresponding `TXT` recordset was created. I've been working on trying to integrate this code for the last couple evenings, and haven't been able to get it working.

**Comment by jeff1evesque** [comment]
My current CloudFormation is a very large set of nested stacks. One component of my build, performs the provisioning of hosted zone(s). I forgot that hosted zones (i.e. name servers) can take up to 48 hours. During that time, the hosted zone(s) will be in `PENDING` state. Kind of clumsy of myself. So, I decoupled this component out of the main CloudFormation nested architecture I have. I will wait until the current state is changed:

```bash
$ ./run --create-hosted-zones
{
    "Location": "https://route53.amazonaws.com/2013-04-01/hostedzone/xxxxxxxxxxxxxxxxxxxxx",
    "HostedZone": {
        "Id": "/hostedzone/xxxxxxxxxxxxxxxxxxxxx",
        "Name": "okcool1.com.",
        "CallerReference": "2020-04-23--22:45:44",
        "Config": {
            "PrivateZone": false
        },
        "ResourceRecordSetCount": 2
    },
    "ChangeInfo": {
        "Id": "/change/aaaaaaaaaaaaaaaaaaaaa",
        "Status": "PENDING",
        "SubmittedAt": "2020-04-24T02:45:46.748000+00:00"
    },
    "DelegationSet": {
        "NameServers": [
            "ns-xxx.awsdns-xx.xxx",
            "ns-xxx.awsdns-xx.xxx",
            "ns-xxx.awsdns-xx.xxx",
            "ns-xxx.awsdns-xx.xxx"
        ]
    }
}


======================================================================

Note: hosted zone(s) can take up to 48 hours to propagate

======================================================================
{
    "Location": "https://route53.amazonaws.com/2013-04-01/hostedzone/yyyyyyyyyyyyyyyyyyyyy",
    "HostedZone": {
        "Id": "/hostedzone/yyyyyyyyyyyyyyyyyyyyy",
        "Name": "okcool2.com.",
        "CallerReference": "2020-04-23--22:45:47",
        "Config": {
            "PrivateZone": false
        },
        "ResourceRecordSetCount": 2
    },
    "ChangeInfo": {
        "Id": "/change/bbbbbbbbbbbbbbbbbbbbb",
        "Status": "PENDING",
        "SubmittedAt": "2020-04-24T02:45:48.768000+00:00"
    },
    "DelegationSet": {
        "NameServers": [
            "ns-xxx.awsdns-xx.xxx",
            "ns-xxx.awsdns-xx.xxx",
            "ns-xxx.awsdns-xx.xxx",
            "ns-xxx.awsdns-xx.xxx"
        ]
    }
}


======================================================================

Note: hosted zone(s) can take up to 48 hours to propagate

======================================================================

$ date
Thu, Apr 23, 2020 10:46:29 PM
```

Then, I will reuse your code verbatim, with `HostedZoneNames` (as you intended), instead of my adjusted `HostedZoneId`. I think this will be more appropriate.

**Comment by jeff1evesque** [comment]
I tried to use the above stated `HostedZoneName`, and it still cannot verify the ses email. The `Pending` state repeats, until the looping is exceeded.

**Comment by jeff1evesque** [comment]
I did some researching, which acknowledged my carelessness. Specifically, I forgot to take the nameserver addresses from my respective hosted zones, and use it to replace the nameserver addresses for my desired domain names(s) in route53. This is likely the issue. However, while I just discovered this to be the issue last night - I've confirmed this by the using the [Binxio](https://github.com/binxio/cfn-certificate-provider) alternative. Their approach utilizes DNS validation. To my knowledge, by utilizing the DNS validation for AWS certificates, the certificates will automatically renew. On the other hand, if certificates were created with email validation, then they will not be eligible for automatic certificate renewal. Nevertheless, I very much appreciate your repository, and all your efforts.
